frontend/app_flowy/packages/flowy_editor/lib/extensions/text_node_extensions.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:flowy_editor/document/node.dart';
+import 'package:flowy_editor/document/path.dart';
+import 'package:flowy_editor/document/position.dart';
+import 'package:flowy_editor/document/selection.dart';
+import 'package:flowy_editor/document/text_delta.dart';
+import 'package:flowy_editor/render/rich_text/rich_text_style.dart';
+
+extension TextNodeExtension on TextNode {
+  bool allSatisfyBoldInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.bold, selection);
+
+  bool allSatisfyItalicInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.italic, selection);
+
+  bool allSatisfyUnderlineInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.underline, selection);
+
+  bool allSatisfyStrikethroughInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.strikethrough, selection);
+
+  bool allSatisfyInSelection(String styleKey, Selection selection) {
+    final ops = delta.operations.whereType<TextInsert>();
+    var start = 0;
+    for (final op in ops) {
+      if (start >= selection.end.offset) {
+        break;
+      }
+      final length = op.length;
+      if (start < selection.end.offset &&
+          start + length > selection.start.offset) {
+        if (op.attributes == null ||
+            !op.attributes!.containsKey(styleKey) ||
+            op.attributes![styleKey] == false) {
+          return false;
+        }
+      }
+      start += length;
+    }
+    return true;
+  }
+}
+
+extension TextNodesExtension on List<TextNode> {
+  bool allSatisfyBoldInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.bold, selection);
+
+  bool allSatisfyItalicInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.italic, selection);
+
+  bool allSatisfyUnderlineInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.underline, selection);
+
+  bool allSatisfyStrikethroughInSelection(Selection selection) =>
+      allSatisfyInSelection(StyleKey.strikethrough, selection);
+
+  bool allSatisfyInSelection(String styleKey, Selection selection) {
```

Comment:
Adding documentation would be better. LOL

Reply:
I will document all the parts of the code next time.